wasm.md

```diff
@@ -980,7 +1022,22 @@ The `#take` function will return the parameter stack in the reversed order, then
            <fAddr> FADDR </fAddr>
            <fType> FTYPE </fType>
            ...
-         </funcDef> requires asFuncType(TYPEIDS, TYPES, TUSE) ==K FTYPE
+         </funcDef> requires unnameFuncType(asFuncType(TYPEIDS, TYPES, TUSE)) ==K unnameFuncType(FTYPE)
+
+    rule <k> call_indirect TUSE:TypeUse => trap ... </k>
+         <curModIdx> CUR </curModIdx>
+         <valstack> < i32 > IDX : VALSTACK => VALSTACK </valstack>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <tabIndices> 0 |-> ADDR </tabIndices>
+           ...
+         </moduleInst>
+         <tabInst>
+           <tAddr> ADDR  </tAddr>
+           <tdata> TDATA </tdata>
+           ...
+         </tabInst>
+      requires notBool IDX in_keys(TDATA)
```

Comment:
Shouldn't it trap if the type is wrong?

Reply:
Yes! I just added the rule for it.
Actually I feel it better to use `=>trap ... [owise]` because the spec says it should trap if anything else happens. But for now, I added the trap if type doesn't match.